addons/project/static/src/js/project_form.js

```diff
@@ -112,10 +95,48 @@ const ProjectFormController = FormController.extend({
     }
 });
 
-export const ProjectFormView = FormView.extend({
+export const FormHtmlFieldExpanderMixin = {
+    bottomDistance: 0,
+    fieldQuerySelector: '.o_xxl_form_view .oe_form_field.oe_form_field_html',
+    on_attach_callback() {
+        this._super(...arguments);
+        this._fixDescriptionHeight();
+    },
+    _fixDescriptionHeight() {
+        if (device.isMobile) return;
+
+        const descriptionField = this.el.querySelector(this.fieldQuerySelector);
+        if (descriptionField) {
+            const editor = descriptionField.querySelector('.note-editable')
+            const elementToResize = editor || descriptionField
+            const minHeight = document.documentElement.clientHeight - elementToResize.getBoundingClientRect().top - this.bottomDistance;
+            elementToResize.style.minHeight = `${minHeight}px`
+        }
+    },
+    _updateView() {
+        this._super(...arguments);
+        this._fixDescriptionHeight();
+    },
+}
+
+const FormDescriptionExpanderRenderer = FormRenderer.extend(Object.assign(FormHtmlFieldExpanderMixin, {
+    // 58px is the sum of the top margin of o_form_sheet 12 px + the bottom padding of o_form_sheet 24px
+    // + 5px margin bottom (o_field_widget) + 1px border + the bottom padding of tab-pane 16 px.
+    bottomDistance: 58,
+}));
+
+export const FormDescriptionExpanderView = FormView.extend({
     config: Object.assign({}, FormView.prototype.config, {
+        Renderer: FormDescriptionExpanderRenderer,
+    }),
+})
+
+export const ProjectFormView = FormDescriptionExpanderView.extend({
+    config: Object.assign({}, FormDescriptionExpanderView.prototype.config, {
         Controller: ProjectFormController,
     }),
 });
 
 viewRegistry.add('project_form', ProjectFormView);
+
+viewRegistry.add('form_description_expander', FormDescriptionExpanderView)
```

Comment:
Missing a EOL and you don't use this view in the XML I think you miss to add `project_` as prefix in the name of the view. 😇 
```suggestion
viewRegistry.add('project_form_description_expander', FormDescriptionExpanderView)
```